.pre-commit-config.yaml

```diff
@@ -37,13 +37,10 @@ repos:
         language: system
         pass_filenames: true
       - id: lint
-        name: revive go lint
-        entry: revive
+        name: golangci-lint go lint
+        entry: golangci-lint
         args:
-          - "-config"
-          - "hack/revive.toml"
-          - "-exclude"
-          - "src/cmd/viper.go"
+          - "run"
```

Comment:
Looks like when we run this on pre-commit it only takes into account the single file as it will run something like `golangci-lint run src/pkg/packager/deploy.go` which can result in false positives like this.
```
src/pkg/packager/deploy.go:35:10: undefined: Packager (typecheck)
func (p *Packager) resetRegistryHPA() {
         ^
src/pkg/packager/deploy.go:44:10: undefined: Packager (typecheck)
func (p *Packager) Deploy() (err error) {
         ^
src/pkg/packager/deploy.go:123:10: undefined: Packager (typecheck)
func (p *Packager) deployComponents() (deployedComponents []types.DeployedComponent, err error) {
```

We should turn off pass_filesiles and use the --new flag imo

Reply:
Took inspiration from another repo and fixed the issue.

https://github.com/golangci/golangci-lint/blob/master/.pre-commit-hooks.yaml